### DISCUSSION TITLE
can() method with multiple guards doesn't work

### DISCUSSION THREAD

**Discussion by Vasiliy-Makogon** [spatie/laravel-permission#2760] [Q&A] [answered]
Version 3.18.0 is installed. Several guards:

```php
    'guards' => [
        'web' => [...],
        'api' => [....],
        'other' => [...],
    ],
```

A middleware is declared:

```php
JsonApi::register('v1')
    ->middleware('auth:api', 'permission:editor,api')
```

This code returns an incorrect value for the `can` method:

```php
var_dump(Auth::guard()->user()->can('editor')); // false ???!!!
var_dump(Auth::guard()->user()->checkPermissionTo('editor', 'api')); // ok - true
```

But if I add this method to the User model:

```php
protected function getDefaultGuardName(): string { return 'api'; }
```

then everything works:

```php
var_dump(Auth::guard()->user()->can('editor')); // ok - true
var_dump(Auth::guard()->user()->checkPermissionTo('editor', 'api')); // ok - true
```

Question: how can I fix the situation and NOT USE the getDefaultGuardName method in the User model?

**Comment by parallels999** [comment, marked as answer]
>Version 3.18.0 is installed. 

Current version is [v6.10.0](https://github.com/spatie/laravel-permission/releases/tag/6.10.0)
```php
var_dump(Auth::guard()->user()->can('editor', 'api')); // try this change
var_dump(Auth::guard()->user()->checkPermissionTo('editor', 'api')); // ok - true
```

**Reply by Vasiliy-Makogon** [reply]
> > Version 3.18.0 is installed.
> 
> Current version is [v6.10.0](https://github.com/spatie/laravel-permission/releases/tag/6.10.0)

The problem is that the dependency `spatie/laravel-permission` in our code comes from a third-party package (https://github.com/BBS-Lab/nova-permission) and I can't predict what will happen to the main package if we update spatie to version 6 :(

**Reply by Vasiliy-Makogon** [reply]
> ```
> try this change
> ```
It didn't help. In general, the global problem is not only that the `can` method doesn't work, because of it the above mentioned intermediaries don't work either:

```php
->middleware('permission:editor,api')
```

Ok. How do I know which version I need to work correctly with Laravel 8? Or is this library supported for all versions of the framework?

**Reply by erikn69** [reply]
>try this change

That only works in the current version

>which version I need to work correctly with Laravel 8?

6.10

**Comment by Vasiliy-Makogon** [comment]
Updated to the latest version and everything worked.
